### PR TITLE
python3Packages.simpy: add missing dependency

### LINKED ISSUE #90494: Unable to build simpy
**Describe the bug**

Building the Python package simpy fails with the following errors:
```
installing 'python3.7-simpy-3.0.13'
these derivations will be built:
  /nix/store/jnfx8gl20jkipbijd44cawjq48k2fyn6-python3.7-simpy-3.0.13.drv
building '/nix/store/jnfx8gl20jkipbijd44cawjq48k2fyn6-python3.7-simpy-3.0.13.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/jzfq1zankfkrmal67j2kq7ymhflzkbkv-simpy-3.0.13.tar.gz
source root is simpy-3.0.13
setting SOURCE_DATE_EPOCH to timestamp 1586095894 of file simpy-3.0.13/src/simpy.egg-info/top_level.txt
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff6b39710>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff6b4c410>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5ee7a90>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5ebd510>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5ebd350>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
ERROR: Could not find a version that satisfies the requirement setuptools_scm (from versions: none)
ERROR: No matching distribution found for setuptools_scm
Traceback (most recent call last):
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/setuptools/installer.py", line 128, in fetch_build_egg
    subprocess.check_call(cmd)
  File "/nix/store/q732h09azy7lf0j30bnnhdl15p4rxpdy-python3-3.7.7/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/q732h09azy7lf0j30bnnhdl15p4rxpdy-python3-3.7.7/bin/python3.7', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmpmx915qy9', '--quiet', 'setuptools_scm']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 40, in <module>
    'Topic :: Scientific/Engineering',
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/setuptools/__init__.py", line 143, in setup
    _install_setup_requires(attrs)
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/setuptools/__init__.py", line 138, in _install_setup_requires
    dist.fetch_build_eggs(dist.setup_requires)
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/setuptools/dist.py", line 698, in fetch_build_eggs
    replace_conflicting=True,
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 783, in resolve
    replace_conflicting=replace_conflicting
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1066, in best_match
    return self.obtain(req, installer)
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1078, in obtain
    return installer(requirement)
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/setuptools/dist.py", line 754, in fetch_build_egg
    return fetch_build_egg(self, req)
  File "/nix/store/ih24yd9fl813ca6h72f9ly0snsvh72zn-python3.7-setuptools-46.1.3/lib/python3.7/site-packages/setuptools/installer.py", line 130, in fetch_build_egg
    raise DistutilsError(str(e))
distutils.errors.DistutilsError: Command '['/nix/store/q732h09azy7lf0j30bnnhdl15p4rxpdy-python3-3.7.7/bin/python3.7', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmpmx915qy9', '--quiet', 'setuptools_scm']' returned non-zero exit status 1.
builder for '/nix/store/jnfx8gl20jkipbijd44cawjq48k2fyn6-python3.7-simpy-3.0.13.drv' failed with exit code 1
error: build of '/nix/store/jnfx8gl20jkipbijd44cawjq48k2fyn6-python3.7-simpy-3.0.13.drv' failed
```
**To Reproduce**

1. Pin yourself to `nixos-unstable` (rev: `"029a5de08390bb03c3f44230b064fd1850c6658a"`, latest available at the moment.)
2. Issue the command: `nix-env -iA nixos.python3Packages.simpy`

**Notify maintainers**

@shlevy 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.6.16, NixOS, 20.09pre228622.029a5de0839 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"nixos-20.09pre228622.029a5de0839"`
 - channels(dmrauh): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
attribute: python3Packages.simpy
```

### PR DESCRIPTION
**Pull request by dmrauh**
###### Motivation for this change

simpy didn't build because of missing `setuptools` as `propagatedBuildInputs` which I added in this PR, thus this resolves #90494. I'm still not to sure about the circumstances which caused the error, but my change fixes it. I also made some smaller cosmetic changes to the derivation while I was at it.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/simpy/default.nix [hunk 1]
```diff
@@ -9,10 +10,17 @@ buildPythonPackage rec {
     sha256 = "dd8c16ca3cff1574c99fe9f5ea4019c631c327f2bdc842e8b1a5c55f5e3e9d27";
   };
 
+  nativeBuildInputs = [ setuptools_scm setuptools ];
```
Comment (jonringer) [inline pkgs/development/python-modules/simpy/default.nix:13]: provide as part of buildPythonPackage
```suggestion
  nativeBuildInputs = [ setuptools_scm ];
```

File: pkgs/development/python-modules/simpy/default.nix [hunk 2]
```diff
@@ -9,10 +10,17 @@ buildPythonPackage rec {
...
+  propagatedBuildInputs = [ setuptools ];
+
+  checkInputs = [ pytestCheckHook ];
+
   meta = with lib; {
-    homepage = "https://simpy.readthedocs.io/en/latest/";
-    description = "A process-based discrete-event simulation framework based on standard Python.";
+    homepage = "https://simpy.readthedocs.io/en/${version}/";
+    description =
+      "A process-based discrete-event simulation framework based on standard Python.";
```
Comment (jonringer) [inline pkgs/development/python-modules/simpy/default.nix:22]: make it fit on one line. And i believe description generally doesn't include a period
```suggestion
    description = "Process-based discrete-event simulation framework based on standard Python";
```

but don't feel strongly about this one.

Comment (dmrauh) [reply pkgs/development/python-modules/simpy/default.nix]: Definitely is more consistent this way!

**Comment by dmrauh** [comment]
Thanks @jonringer! The PR should be ready now.

File: pkgs/development/python-modules/simpy/default.nix [hunk 3]
```diff
@@ -1,4 +1,5 @@
-{ buildPythonPackage, fetchPypi, lib, setuptools_scm, pytestCheckHook }:
+{ buildPythonPackage, fetchPypi, lib, setuptools, setuptools_scm
+, pytestCheckHook }:
 
 buildPythonPackage rec {
   pname = "simpy";
```
Comment (jonringer) [inline pkgs/development/python-modules/simpy/default.nix:5]: should be added after version, just can't see in github diff
```suggestion
  pname = "simpy";
  disabled = isPy27;
```

```
builder for '/nix/store/vhac4469zyak4w2ab7s2w1vyql8k1rpc-python2.7-simpy-4.0.1.drv' failed with exit code 1; last 10 log lines:
  adding 'simpy-4.0.1.dist-info/top_level.txt'
  adding 'simpy-4.0.1.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Finished executing setuptoolsBuildPhase
  installing
  Executing pipInstallPhase
  /build/simpy-4.0.1/dist /build/simpy-4.0.1
  DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
  Processing ./simpy-4.0.1-py2.py3-none-any.whl
  ERROR: Package 'simpy' requires a different Python: 2.7.18 not in '>=3.6'
```

**Review by jonringer — COMMENTED** [review]
please squash commits, should just be one :)

File: pkgs/development/python-modules/simpy/default.nix (hunk 3 above)
Comment (dmrauh) [reply pkgs/development/python-modules/simpy/default.nix:5]: I've disabled Python 2 compatibility, since Python 3.6 is the minimum supported version.

**Comment by dmrauh** [comment]
Squashed the commits :)

**Comment by Ma27** [comment]
It builds for me on `master` (09aa2893580ec6dcadf5a87b89e4d4c6f4ad888e): `/nix/store/dqbs7hr2cray5kbw0ky7brs9sm55xar5-python3.7-simpy-4.0.1`

Also it's possible to import the lib in a `python3` session.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 90598 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  - python37Packages.simpy  - python38Packages.simpy</details>

**Comment by Ma27** [comment]
@Jonringer was it broken for you as well?

**Comment by jonringer** [comment]
```
[16:43:50] jon@jon-desktop /home/jon/projects/nixpkgs (master)
$ nix-shell -p "with import ./. {}; python3Packages.simpy"

[nix-shell:/home/jon/projects/nixpkgs]$ python
Python 3.8.3 (default, May 13 2020, 19:59:26)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import simpy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/bs3ld6xhn0s8nmai93wnw55simisir33-python3.8-simpy-4.0.1/lib/python3.8/site-packages/simpy/__init__.py", line 11, in <module>
    from pkg_resources import get_distribution
ModuleNotFoundError: No module named 'pkg_resources'
```

**Comment by jonringer** [comment]
@Ma27 setuptools is a pretty common package to require implicitly.

**Comment by dmrauh** [comment]
Thanks for merging!

**Pull request merged**